### PR TITLE
Resolve "Move Spinner Task into Apps Directory"

### PR DESCRIPTION
**Pull request by Kfeavel**
Need to figure out how to get both to print to the screen at the same time before merging.

Also fix a typo in the isr_install.

### CODE REVIEW

File: kernel/apps/primes.cpp [hunk 1]
```diff
@@ -13,7 +23,7 @@ static bitmap_t primes[PRIMES_SIZE];
 
 static size_t prime_current;
 
-void find_primes(void)
+void task_find_primes(void)
```
Comment (micahswitzer) [inline kernel/apps/primes.cpp]: Namespaces?

File: kernel/apps/primes.cpp [hunk 2]
```diff
@@ -23,11 +33,11 @@ void find_primes(void)
         for (size_t j = prime_current * prime_current; j < PRIME_MAX; j += prime_current) {
             bitmap_clear_bit(primes, j);
         }
-        tasks_nano_sleep(1000ULL * 1000 * 100);
+        //tasks_nano_sleep(1000ULL * 1000 * 100);
```
Comment (micahswitzer) [inline kernel/apps/primes.cpp]: Maybe add a note about why that was there to begin with. Otherwise, just delete the line.

File: kernel/main.cpp [hunk 3]
```diff
@@ -77,39 +78,27 @@ void kernel_main(void *boot_info, uint32_t magic) {
...
-    // Done
-    kprintf(DBG_OKAY "Done.\n");
-
     tasks_init();
     task_t compute, status;
-    tasks_new(find_primes, &compute, TASK_READY, "prime_compute");
-    tasks_new(show_primes, &status, TASK_READY, "prime_display");
+    tasks_new(task_find_primes, &compute, TASK_READY, "prime_compute");
+    tasks_new(task_show_primes, &status, TASK_READY, "prime_display");
+    //tasks_new(task_spinner, &status, TASK_READY, "spinner");
```
Comment (micahswitzer) [inline kernel/main.cpp]: Make sure you create a new `task_t` for the spinner.

File: kernel/apps/primes.cpp (hunk 2 above)
Comment (Kfeavel) [reply kernel/apps/primes.cpp]: Ah, yeah, meant to just remove it.

File: kernel/main.cpp (hunk 3 above)
Comment (Kfeavel) [reply kernel/main.cpp]: That fixed the printing issue. Good catch.

File: kernel/apps/primes.cpp (hunk 1 above)
Comment (Kfeavel) [reply kernel/apps/primes.cpp]: Good idea.

Comment (Kfeavel) [reply kernel/apps/primes.cpp]: Resolved in b246794.

File: kernel/main.cpp (hunk 3 above)
Comment (Kfeavel) [reply kernel/main.cpp]: Resolved in b246794.

File: kernel/apps/primes.cpp (hunk 2 above)
Comment (Kfeavel) [reply kernel/apps/primes.cpp]: Resolved in b246794.

File: kernel/apps/primes.cpp (hunk 1 above)
Comment (micahswitzer) [reply kernel/apps/primes.cpp]: Oh well I was thinking we wouldn't need the `task_` portion if we used namespaces, but I guess I'm fine with whatever

File: kernel/main.cpp [hunk 4]
```diff
@@ -77,39 +78,27 @@ void kernel_main(void *boot_info, uint32_t magic) {
...
-    int i = 0;
-    const char spinnay[] = { '|', '/', '-', '\\' };
-    while (true) {
-        // Display a spinner to know that we're still running.
-        kprintf("\e[s\e[24;0f%c\e[u", spinnay[i]);
-        i = (i + 1) % sizeof(spinnay);
-        asm volatile("hlt");
-    }
+    // Keep the kernel task alive.
+    tasks_block_current(TASK_SLEEPING);
```
Comment (micahswitzer) [inline kernel/main.cpp]: Technically the correct blocking reason would be `TASK_PAUSED`.

File: sysroot/usr/include/kernel/apps/primes.hpp [hunk 5]
```diff
@@ -1,4 +1,27 @@
+/**
+ * @file primes.hpp
+ * @author Micah Switzer (mswitzer@cedarville.edu)
+ * @brief Prime computation tasks
+ * @version 0.1
+ * @date 2021-06-05
```
Comment (micahswitzer) [inline sysroot/usr/include/kernel/apps/primes.hpp]: I know you just added these headers, but would it be worth it to put the original author date of these files?

Comment (Kfeavel) [reply sysroot/usr/include/kernel/apps/primes.hpp]: Sure, I can do that.

File: kernel/main.cpp (hunk 4 above)
Comment (Kfeavel) [reply kernel/main.cpp]: Resolved in a5c787b.

File: sysroot/usr/include/kernel/apps/primes.hpp (hunk 5 above)
Comment (Kfeavel) [reply sysroot/usr/include/kernel/apps/primes.hpp]: Resolved in a5c787b.

File: kernel/apps/primes.cpp (hunk 1 above)
Comment (Kfeavel) [reply kernel/apps/primes.cpp]: Resolved in a5c787b.

**Review by micahswitzer — APPROVED** [review]

**Pull request merged**